tests/golos.publication_rewards_tests.cpp

```diff
@@ -880,4 +910,57 @@ BOOST_FIXTURE_TEST_CASE(golos_curation_test, reward_calcs_tester) try {
     check();
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(golos_delegators_test, reward_calcs_tester) try {
+    BOOST_TEST_MESSAGE("golos_delegators_test");
+    int64_t maxfp = std::numeric_limits<fixp_t>::max();
+    auto bignum = 500000000000;
+    init(bignum, 500000);
+    produce_blocks();
+    BOOST_TEST_MESSAGE("--- setrules");
+    using namespace golos_curation;
+    BOOST_CHECK_EQUAL(success(), setrules({"x", maxfp}, {golos_curation::func_str, maxfp}, {"1", bignum},
+        [](double x){ return x; }, golos_curation::func, [](double x){ return 1.0; }));
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+
+    BOOST_TEST_MESSAGE("--- create_message: " << name{_users[0]}.to_string());
+    BOOST_CHECK_EQUAL(success(), create_message({_users[0], "permlink"}));
+    check();
+
+    const uint8_t withdraw_intervals = 1;
+    const uint32_t withdraw_interval_seconds = 12;
+    const uint64_t vesting_min_amount = 0;
+    const uint64_t delegation_min_amount = 5e6;
+    const uint64_t delegation_min_remainder = 15e3;
+    const uint32_t delegation_min_time = 0;
+    const uint16_t delegation_max_interest = 5000;
+    const uint32_t delegation_return_time = 120;
+
+    auto vesting_withdraw = vest.withdraw_param(withdraw_intervals, withdraw_interval_seconds);
+    auto vesting_amount = vest.amount_param(vesting_min_amount);
+    auto delegation = vest.delegation_param(delegation_min_amount, delegation_min_remainder, delegation_min_time, delegation_max_interest, delegation_return_time);
+
+    auto params = "[" + vesting_withdraw + "," + vesting_amount + "," + delegation + "]";
+    BOOST_CHECK_EQUAL(success(), vest.set_params(_issuer, _token_symbol, params));
+
+    BOOST_TEST_MESSAGE("--- delegate_vesting");
+    const int divider = vest.make_asset(1).get_amount();
+    const auto min_fract = 1. / divider;
+    const int min_remainder = delegation_min_remainder / divider;
+    const auto amount = vest.make_asset(min_remainder);
+    const auto charge_prop = 0.7;
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[2], _users[1], amount, 3000, 0));
+    check();
+
+    BOOST_TEST_MESSAGE("--- " << name{_users[1]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[1], {_users[0], "permlink"}, 10000));
+    check();
```

Comment:
this is not enough to be sure delegators payments work. there can be errors in both model & contract, so we should manually check delegator's and curator's balance (at least check it changes (increases) when it should or not changes when it shouldn't).

Also here only 1 case with only 1 delegator. Please add checks for the following cases:
1. one delegator, 100% interest, 0% interest
2. several delegators, each gets payment
3. several delegators, one (with non-zero interest rate) gets `continue` here: https://github.com/GolosChain/golos.contracts/blob/8aa8ad21358f9d83d041e2f4f85a11d7fb021ad2/tests/golos.publication_rewards_tests.cpp#L388 (please use constants in test so it's clear why expecting value becomes less than min percent. e.g. interest rate is 10%, so `dlg.amount / effective_vesting` should be less than 0.1)